api/src/apollo/Event/queries.ts

```diff
@@ -1,54 +1,39 @@
-import { idArg, list, nonNull, queryField } from 'nexus';
+import { checkGqlPermissions } from 'helpers/auth/checkPermissions';
+import { idArg, list, queryField } from 'nexus';
 
-export const getEvent = queryField('getEvent', {
-  type: 'Event',
-  description: `Returns the event whose id is passed as an argument`,
-  args: {
-    id: nonNull(idArg()),
-  },
-  resolve(_parent, args, { prisma }) {
-    return prisma.event.findUnique({
-      where: {
-        id: args.id,
-      },
-    });
-  },
-});
-
-export const getEvents = queryField('getEvents', {
+export const event = queryField('event', {
   type: list('Event'),
-  description: `Returns a list of events depending upon the various arguments passed`,
+  description: `Returns as list of events depending upon the arguments`,
+  authorize: (_parent, args, ctx) =>
+    args.id || args.orgID ? true : checkGqlPermissions(ctx, []),
```

Comment:
I believe getEvents should be open na?

Reply:
no it should not. If it is open for all, anyone can fetch each and every possible event (draft, other organisations, etc) which we don't really want.